tests/unit/lib/observers/ParameterObserverTest_unittest.cc

```diff
@@ -138,6 +142,18 @@ TEST_F(ParameterObserverTest, filter_all_property)
 	emitter.subscribe(observer);
 	emitter.emit_value();
 
-	EXPECT_EQ(observer->get<int32_t>("num_observations"), 4);
+	EXPECT_EQ(observer->get<int32_t>("num_observations"), 5);
 	emitter.unsubscribe(observer);
+}
+
+TEST_F(ParameterObserverTest, filter_method)
+{
+	std::vector<std::string> method_params = {"method"};
+	std::shared_ptr<ParameterObserver> observer(
+	    new ParameterObserverLogger(method_params));
+	emitter.subscribe(observer);
+	emitter.emit_value();
+
+	EXPECT_EQ(observer->get<int32_t>("num_observations"), 1);
+    EXPECT_EQ(observer->get_observation(0)->get_any().as<int32_t>(), 10);
```

Comment:
nice.
Another test should be to test that the method is not called as written above